base/c.jl

```diff
@@ -503,3 +503,183 @@ end
 macro ccallable(rt, def)
     expand_ccallable(rt, def)
 end
+
+module CcallMacro
+
+struct NoType end
+struct CcallError{T <: AbstractString} <: Exception
+    msg::T
+end
+Base.showerror(io::IO, e::CcallError) = print(io, "CcallError: ", e.msg)
+
+getcfunc(f) = QuoteNode(f)
+getcfunc(f::Expr) = :(($(f.args[2]), $(f.args[1])))
+hashead(symbol, _) = false
+hashead(expr::Expr, head) = expr.head === head
```

Comment:
I often use dispatch instead of conditional expressions if it helps reduce compound boolean expressions--the way one would use pattern matching on algebraic data types in ML/Haskell/Rust.

However, if this is considered an anti-pattern in Julia, I would be glad to change my approach.

Reply:
That's generally fine. This is just already the `Meta.isexpr` function, and we try not to duplicate too much

Reply:
fixed in https://github.com/JuliaLang/julia/pull/32748/commits/13c8bbf5db6adb4c43244a4b23ce6d4e4de42336

Reply:
Agreed — `hashead` is a good idea; it's just that we already have something which does this.